llvm/include/llvm/Object/DXContainer.h

```diff
@@ -120,9 +120,10 @@ template <typename T> struct ViewArray {
 namespace DirectX {
 struct RootParameterView {
   const dxbc::RootParameterHeader &Header;
+  uint32_t Version;
   StringRef ParamData;
-  RootParameterView(const dxbc::RootParameterHeader &H, StringRef P)
-      : Header(H), ParamData(P) {}
+  RootParameterView(uint32_t V, const dxbc::RootParameterHeader &H, StringRef P)
+      : Header(H), Version(V), ParamData(P) {}
 
   template <typename T> Expected<T> readParameter() {
```

Comment:
```suggestion
  template <typename T, typename VersionT = T> Expected<T> readParameter() {
    assert(sizeof(VersiontT) <= sizeof(T) && "Parameter of higher version must inherit all previous version data members");
```
and then below we would just copy the relevant data members:
`memcpy(&Struct, ParamData.data(), sizeof(VersionT))`

Reply:
Great suggestion, thanks